js/exportsms.js

```diff
@@ -417,7 +411,116 @@ var exportsms = function() {
+      "group-level-success-oips": [],
+      "group-success-failure-oips": {
+        "0": null,
+        "1": null,
+        "2": null,
+        "3": null,
+        "4": null,
+        "5": null,
+        "6": null
+      },
+      "indiv-level-success-oips": [],
+      "indiv-rank-oips": {
+        "1": null,
+        "2": null,
+        "3": null,
+        "4": null,
+        "1-tied": null,
+        "2-tied": null
+      }
+    };
+
+    // Iterating through all the endgame data.
+    for (j = 0; j < endGameData.length; j++) {
+      endGameDatum = endGameData[j];
+      // Storing oip for an individual rank result.
+      if (endGameDatum.type === 'passageEndGameIndivRankResult') {
+        rank = endGameDatum.rank;
+        oip = endGameDatum.optinpath;
+        endGameObject["indiv-rank-oips"][rank] = oip;
+      }
+      // Storing oip for an individual superlative result, and setting up the logic choice object. 
+      else if (endGameDatum.type === 'passageEndGameIndivSuperlativeResult') {
+        superlative = endGameDatum.pathflag;
+        for (k = 0; k < endGameObject.choices.length; k++) {
+          if (endGameObject.choices[k].flag == superlative) {
+            choiceObject = endGameObject.choices[k];
+          }
+        }
+        // Checking that choiceObject doesn't exist or that the namespace isn't already taken by something irrelevant. 
+        if (!choiceObject || (choiceObject.flag != superlative)) {
+          choiceObject = {
+            "next" : '',
+            "flag" : '',
+            "conditions" : {
+              "$and": []
+            }
+          }
+        }
+        choiceObject.next = endGameDatum.optinpath;
+        choiceObject.flag = superlative;
+        endGameObject.choices.push(choiceObject);
+      }
+
+      // Retrieving oip for a group success number result. 
+      else if (endGameDatum.type === 'passageEndGameGroupSuccessNumberResult') {
+        min = parseInt(endGameDatum.minnumlevelsuccess);
+        max = parseInt(endGameDatum.maxnumlevelsuccess);
+        for (l = min; l <= max; l++) {
+          endGameObject["group-success-failure-oips"][l] = endGameDatum.optinpath;
+        }
+      }
+    }
+
+    // Iterating through all the endlevel data. 
+    for (i = 0; i < endLevelData.length; i++) {
+      endLevelDatum = endLevelData[i];
+      if (endLevelDatum.type === 'passageEndLevelIndiv') {
+        // Storing oip of an individual success level. Used to calculate endgame result of rank among others. 
+        if (endLevelDatum.indivsuccesspath == 'true') {
+          endGameObject["indiv-level-success-oips"].push(endLevelDatum.optinpath);
+        }
+        // Storing the key of a passage which a user must traverse in order to receive a specific superlative endgame result. 
+        if (endLevelDatum.indivsuperlativepath) {
+          superlativeArray = endLevelDatum.indivsuperlativepath.split(/[ ,]+/);
+          for (p = 0; p < superlativeArray.length; p++) {
```

Comment:
Ahh--so I originally named the loops in order of: i, j, k, l, m, n, q ,p. 

Some of the loops, however, got rearranged. I can re-order the variables, though, if it impacts readability?